subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1818,13 +1818,17 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t remainder, uint16_t laz
 			uint32_t ticks_interval =
 				HAL_TICKER_US_TO_TICKS((uint64_t)adv->interval *
 						       ADV_INT_UNIT_US);
-			if (adv->ticks_remain_duration > ticks_interval) {
-				adv->ticks_remain_duration -= ticks_interval;
+			uint32_t ticks_elapsed = ticks_interval * (lazy + 1);
+
+			if (adv->ticks_remain_duration > ticks_elapsed) {
+				adv->ticks_remain_duration -= ticks_elapsed;
```

Comment:
If the interval is larger than `ticks_remaing_duration` after this reduction, shouldn't we stop advertising already now? I.e. if we know we cannot schedule a new advertisement?

Reply:
Duration can be 10ms, and interval 20 ms. In this case, first radio event should still be transmitted and then stopped. Hence, before transmission the decrement is done and the termination is done after the transmission